Time system/Assets/Code/Scripts/Environment/DayNightCycle.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using Code.Scripts.Enums;
+using Code.Scripts.Environment.TimeSystem;
+using Code.Scripts.ScriptableObjects;
+using UnityEngine;
+
+namespace Code.Scripts.Environment
+{
+    [ExecuteAlways]
+    public class DayNightCycle : MonoBehaviour
+    {
+        private const float SunInitialIntensity = 1;
+
+        [SerializeField]
+        private TimeSettings timeSettings;
+
+        [SerializeField]
+        private TimeManager timeManager;
+
+        [SerializeField]
+        private Light sunLight;
+
+        [SerializeField]
+        private DayNightSettings dayNightSettings;
+
+        [SerializeField]
+        [Range(0, 24)]
+        private float timeOfDay;
+
+        // Internal use
+        private float internalTimeOfDay;
+
+        private float intensityMultiplier = .8f;
+
+        private void Start()
+        {
+            if (sunLight != null && RenderSettings.sun != null)
+            {
+                sunLight = RenderSettings.sun;
+
+                if (sunLight == null)
+                {
+                    Debug.LogError("Expected a minimum of 1 (directional) light in the scene");
+                }
+            }
+        }
+
+        private void Update()
+        {
+            if (Application.isPlaying)
+            {
+                internalTimeOfDay = (timeManager.Ticks / (int) TimeUnits.SECONDS_IN_DAY) % 1;
+            }
+            else
+            {
+                internalTimeOfDay = timeOfDay / (float) TimeUnits.HOURS_IN_DAY;
+            }
+
+            if (internalTimeOfDay <= Hour(5.52f) || internalTimeOfDay >= Hour(18))

```

Comment:
Hier staan magic numbers. Deze lijken mij ook waardes die het gameplay team willen regelen. (Dit is het moment van het begin en eind van de dag toch?)

Reply:
Eerder het begin en einde van de ochtend/middag/nacht/etc. Deze waardes zijn nu instelbaar.